src/lib.rs

```diff
@@ -74,9 +74,10 @@ pub trait ThreadTracer {
 }
 
 /// Start tracing on the current thread using the specified tracing kind.
-/// A thread tracer is returned through which tracing can be stopped and the trace obtained. The
-/// user must ensure that only one thread tracer exists at any one time per-thread, otherwise
-/// undefined behaviour will result.
+/// By passing `kind` as `None` an appropriate tracing kind will be selected automatically. A
```

Comment:
How about simplifying to something like "If `None` is passed, then an appropriate backend will be selected; if `Some(...)` is passed, a specific backend can be specified. Only one thread tracer can be started at any point; starting more than one leads to undefined behaviour."?

Reply:
Try my last push. I want the restrictions on threads to be super-clear.